picard/ui/searchdialog.py

```diff
@@ -0,0 +1,400 @@
+            self.save_state(True)
+        else:
+            self.save_state(False)
+
+        QtGui.QDialog.reject(self)
+
+    def restore_state(self):
+        size = config.persist["searchdialog_window_size"]
+        if size:
+            self.resize(size)
+        self.search_box.restore_checkbox_state()
+
+    def restore_table_header_state(self):
+        header = self.table.horizontalHeader()
+        state = config.persist["searchdialog_header_state"]
+        if state:
+            header.restoreState(state)
+        header.setResizeMode(QtGui.QHeaderView.Interactive)
+
+    def save_state(self, table_loaded=True):
+        if table_loaded:
+            header = self.table.horizontalHeader()
+            config.persist["searchdialog_header_state"] = header.saveState()
+        config.persist["searchdialog_window_size"] = self.size()
+        self.search_box.save_checkbox_state()
+
+
+class TrackSearchDialog(SearchDialog):
+
+    def __init__(self, parent):
+        super(TrackSearchDialog, self).__init__(parent)
+        self.file_ = None
+        self.setWindowTitle(_("Track Search Results"))
+        self.table_headers = [
+                _("Name"),
+                _("Length"),
+                _("Artist"),
+                _("Release"),
+                _("Date"),
+                _("Country"),
+                _("Type")
+        ]
+
+    def search(self, text):
+        self.retry_params = (self.search, text)
+        self.search_box.search_edit.setText(text)
+        self.show_progress()
+        self.tagger.xmlws.find_tracks(self.handle_reply,
+                query=text,
+                search=True,
+                limit=25)
+
+    def load_similar_tracks(self, file_):
+        self.retry_params = (self.load_similar_tracks, file_)
+        self.file_ = file_
+        metadata = file_.orig_metadata
+        query = {
+                'track': metadata['title'],
+                'artist': metadata['artist'],
+                'release': metadata['album'],
+                'tnum': metadata['tracknumber'],
+                'tracks': metadata['totaltracks'],
+                'qdur': str(metadata.length / 2000),
+                'isrc': metadata['isrc'],
+        }
+        if config.setting["use_adv_search_syntax"]:
+            query_str = ' '.join(['%s:(%s)' % (item, value) for item, value in query.iteritems() if value])
+        else:
+            query_str = query["track"]
+        query["limit"] = 25
+        self.search_box.search_edit.setText(query_str)
+        self.show_progress()
+        self.tagger.xmlws.find_tracks(
+                self.handle_reply,
+                **query)
+
+    def retry(self):
+        self.retry_params[0](self.retry_params[1])
+
+
+    def handle_reply(self, document, http, error):
+        if error:
+            error_msg = _("Some network error occurred. Check debug logs for more details.<br>"
+                "Hit `Retry` or try a different query."
```

Comment:
This should be using either ASCII double quotes (`"Retry"`) or Unicode ones (`“Retry”`).